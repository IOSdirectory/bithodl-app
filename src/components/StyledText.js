import React, { Component } from 'react';
import { StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  text: {
    color: '#504F63',
    fontFamily: 'Avenir Next',
    fontWeight: '400'
  }
});

export default class StyledText extends Component {
  render() {
    return (
      <Text style={[styles.text, this.props.style]} numberOfLines={this.props.numberOfLines}>
        {this.props.children}
      </Text>
    );
  }
}

StyledText.propTypes = {
  style: PropTypes.any,
  numberOfLines: PropTypes.number,
  children: PropTypes.node
};
