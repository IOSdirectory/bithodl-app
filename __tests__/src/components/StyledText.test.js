import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import StyledText from '../../../src/components/StyledText';

describe('StyledText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <StyledText numberOfLines={1}>
        <Text>56d8ab7b-d089-46e7-aefc-526768cd08a6</Text>
      </StyledText>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
