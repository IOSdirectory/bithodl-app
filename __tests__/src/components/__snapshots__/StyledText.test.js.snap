// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Object {
        "color": "#504F63",
        "fontFamily": "Avenir Next",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    56d8ab7b-d089-46e7-aefc-526768cd08a6
  </Text>
</Text>
`;
